tools/ci/Dockerfile

```diff
@@ -1,29 +1,31 @@
-FROM ubuntu:18.04
-
+FROM ubuntu:18.04 as builder
 LABEL maintainer service@fisco.com.cn
 
-ARG SOURCE_BRANCH
-ENV SOURCE=${SOURCE_BRANCH:-master}
-
 WORKDIR /
 
+ARG SOURCE_BRANCH
+ENV DEBIAN_FRONTEND=noninteractive \
+    SOURCE=${SOURCE_BRANCH:-master}
+
 RUN apt-get -q update && apt-get install -qy --no-install-recommends \
     git clang make build-essential cmake libssl-dev zlib1g-dev ca-certificates \
-    && export DEBIAN_FRONTEND=noninteractive \
     && ln -fs /usr/share/zoneinfo/Asia/Shanghai /etc/localtime \
     && apt-get install -qy --no-install-recommends tzdata \
     && dpkg-reconfigure --frontend noninteractive tzdata \
-    && git clone https://github.com/FISCO-BCOS/FISCO-BCOS.git \
-    && cd FISCO-BCOS && git checkout ${SOURCE} \
-    && cmake . && make && make install \
-    && cd / && rm -rf FISCO-BCOS \
-    && apt-get purge git cmake build-essential -y \ 
-    && apt-get autoremove -y \
-    && apt-get clean \
-    && rm  -rf /tmp/* \
     && rm -rf /var/lib/apt/lists/*
 
+RUN git clone https://github.com/FISCO-BCOS/FISCO-BCOS.git && cd FISCO-BCOS \
+    && git checkout ${SOURCE} && cmake . && make && strip bin/fisco-bcos
+
+FROM ubuntu:18.04
+LABEL maintainer service@fisco.com.cn
+
+RUN apt-get -q update && apt-get install -qy --no-install-recommends libssl-dev zlib1g-dev
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5318653)